app/controllers/crypto_address_for_channels_controller.rb

```diff
@@ -0,0 +1,102 @@
+require "eth"
+require "rbnacl"
+require "base58"
+
+class CryptoAddressForChannelsController < ApplicationController
+  include Eth
+  include ChannelsHelper
+
+  before_action :authenticate_publisher!
+  before_action :setup_current_channel, except: :generate_nonce
+
+  def index
+    @crypto_addresses_for_channel = CryptoAddressForChannel.where(channel: @current_channel)
+
+    render(json: @crypto_addresses_for_channel)
+  end
+
+  def generate_nonce
+    render json: {nonce: SecureRandom.hex(32)}
+  end
+
+  def create
+    signature = params[:transaction_signature]
+    account_address = params[:account_address]
+    chain = params[:chain]
+    message = params[:message]
```

Comment:
We should make sure that the nonce we're generating and sending to the front end matches on verification here as well as checking to make sure it's the same user

Reply:
As I understand it `message` is supposed to be just the nonce. However, we are not checking anything about `message` so it could even be the empty string as long as that is signed correctly.

We should verify that the nonce (`message`) was generated by our backend and that it is not being reused.
Brave Talk, for instance, used Redis. The nonce generation endpoint SETs a key for the nonce and verification does the equivalent of GETDEL to check that it is a valid nonce and delete it so it cannot be reused.